src/sink/gpub.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # File Offramp
+//!
+//! Writes events to a file, one event per line
+//!
+//! ## Configuration
+//!
+//! See [Config](struct.Config.html) for details.
+
+#![cfg(not(tarpaulin_include))]
+
+use crate::connectors::gcp::{pubsub, pubsub_auth};
+use crate::connectors::qos::{self, QoSFacilities, SinkQoS};
+use crate::sink::prelude::*;
+use futures::executor::block_on;
+use googapis::google::pubsub::v1::publisher_client::PublisherClient;
+use halfbrown::HashMap;
+use tonic::transport::Channel;
+use tremor_pipeline::{EventIdGenerator, OpMeta};
+use tremor_value::Value;
+
+pub struct GoogleCloudPubSub {
+    #[allow(dead_code)]
```

Comment:
we should try to avoid allows

Reply:
complains about the `config field not being read, if I remove this allow` can I add a comment for this too?

Reply:
We could delete it if it's not used :)